client/my-sites/themes/controller/index.node.js

```diff
@@ -42,27 +40,14 @@ export function fetchThemeDetailsData( context, next ) {
 	}
 
 	const themeSlug = context.params.slug;
-	const theme = themeDetailsCache.get( themeSlug );
+	const theme = getThemeDetails( context.store.getState(), themeSlug );
 
-	if ( theme ) {
+	if ( Object.keys( theme ).length ) {
```

Comment:
Alternatively, we could add a call to `fetchThemeDetails` in this branch (without the callback), so that the cache is always up to date.